dimod/sampleset.py

```diff
@@ -1450,17 +1449,33 @@ def relabel_variables(self, mapping, inplace=True):
             Variables([0, 1])
 
         """
-        if not inplace:
+        done = self.done()
+
+        if inplace and done:
+            self.variables._relabel(mapping)
+            return self
+
+        elif done:  # and not inplace
             return self.copy().relabel_variables(mapping, inplace=True)
 
-        if not self.done():
+        elif inplace:  # and not done
+            old_hook = self._result_hook
+
+            def new_hook(future):
+                sampleset = old_hook(future)
```

Comment:
Why define `old_hook` rather than just use ` sampleset = self._result_hook(future)` ?

Reply:
That creates a copy (and was the bugged pattern it was using before).